proxy/proxy_handlers.go

```diff
@@ -0,0 +1,112 @@
+package proxy
+
+import (
+	"bufio"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/websocket"
+	"github.com/taskcluster/webhooktunnel/util"
+	"github.com/taskcluster/webhooktunnel/wsmux"
+)
+
+func (p *proxy) handler(w http.ResponseWriter, r *http.Request) {
+	// register a worker
+	p.logger.Printf("new request: %s, Method: %s", r.URL.Path, r.Method)
+	if strings.HasPrefix(r.URL.Path, "/register") {
+		p.register(w, r)
```

Comment:
Don't read URL here and then pass `w, r`... if you read URL here you should parse it completely and pass the values like `ID` on to the register...

By all means avoid parsing the URL twice, it's a great source of bugs...